### PR TITLE
Get the deployment flow working with GCE

### PR DESCRIPTION
**Pull request by huwshimi**

### CODE REVIEW

File: jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js [hunk 1]
```diff
@@ -66,7 +66,11 @@ YUI.add('deployment-credential-add', function() {
       const info = this._getInfo();
       const fields = {};
       info.forms[this.state.authType].forEach(field => {
-        fields[field.id] = this.refs[field.id].getValue();
+        let value = this.refs[field.id].getValue();
+        if (field.unescape) {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js]: Needs tests.

File: jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js [hunk 2]
```diff
@@ -183,11 +187,25 @@ YUI.add('deployment-credential-add', function() {
       const fields = info.forms[this.state.authType].map(field => {
         if (field.json) {
           return (
-            <div className="deployment-credential-add__upload twelve-col"
+            <div className="deployment-credential-add__upload"
               key={field.id}>
-              Upload {info.title} auth-file.
+              <juju.components.FileField
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js]: Needs tests.

File: jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js [hunk 3]
```diff
@@ -183,11 +187,25 @@ YUI.add('deployment-credential-add', function() {
...
+                accept=".json"
+                disabled={isReadOnly}
+                key={field.id}
+                label={`Upload ${info.title} .json auth-file`}
+                required={true}
+                ref={field.id} />
             </div>);
         }
+        if (field.multiLine) {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/deployment-flow/credential/add/add.js]: Needs tests.

File: jujugui/static/gui/src/app/components/file-field/file-field.js [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    focus: function() {
+      return this.refs.field.focus();
+    },
+
+    /**
+      Handle uploading the file and getting the value.
+
+      @method _handleFileChange
+    */
+    _handleFileChange: function() {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/file-field/file-field.js]: Needs tests.

File: jujugui/static/gui/src/app/components/string-config/string-config.js [hunk 5]
```diff
@@ -50,6 +50,15 @@ YUI.add('string-config', function() {
     },
 
     /**
+      Get the value of the field.
+
+      @method getValue
+    */
+    getValue: function() {
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/string-config/string-config.js]: Needs tests.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5909/
Test FAILed.

**Comment by hatched** [comment]
Closing because additional work has been done on this, creating a new PR

**Pull request closed without merging**